python/tk_multi_loader/model_item_data.py

```diff
@@ -0,0 +1,188 @@
+    #      'path':            '/Project/70/Asset/sg_asset_type/Character',
+    #      'ref':             {'kind': 'list', 'value': 'Character'},
+    #      'target_entities': {'additional_filter_presets': [{'path':        '/Project/70/Asset/sg_asset_type/Character',
+    #                                                         'preset_name': 'NAV_ENTRIES',
+    #                                                         'seed':        {'field': 'entity', 'type': 'PublishedFile'}}],
+    #                          'type': 'PublishedFile'}}
+    #
+    # Example for a ShotgunHierarchyModel leaf item (with the extra entity fields requested):
+    #     {'has_children':    False,
+    #      'label':           'Bunny',
+    #      'path':            '/Project/70/Asset/sg_asset_type/Character/id/1230',
+    #      'ref':             {'kind': 'entity', 'value': {'id':             1230,
+    #                                                      'type':           'Asset',
+    #                                                      'code':           'Bunny',
+    #                                                      'description':    '...',
+    #                                                      'image':          '/thumbnail/Asset/1230?567',
+    #                                                      'sg_status_list': 'fin'}},
+    #      'target_entities': {'additional_filter_presets': [{'path':        '/Project/70/Asset/sg_asset_type/Character/id/1230',
+    #                                                         'preset_name': 'NAV_ENTRIES',
+    #                                                         'seed':        {'field': 'entity', 'type': 'PublishedFile'}}],
+    #                          'type': 'PublishedFile'}}
+    #
+    sg_data = shotgun_model.get_sg_data(item)
+
+    # Get the item data for user role SG_ASSOCIATED_FIELD_ROLE.
+    #
+    # For an item in the ShotgunModel tree structure,
+    # this data is a dictionary with keys "name" and "value":
+    # - for an entity link, this value is a dictionary with keys:
+    #       "id", "name" and "type"
+    # - othewise, this value is the text data.
+    # - "value" can also be a list of such values.
+    #
+    # Examples for a ShotgunModel intermediate item:
+    #     {'name': 'sg_asset_type', 'value': 'Character'}
+    #     {'name': 'sg_sequence', 'value': {'id': 23, 'name': 'bunny_010', 'type': 'Sequence'}}
+    #
+    # Examples for a ShotgunModel leaf item:
+    #     {'name': 'code', 'value': 'Bunny'}
+    #     {'name': 'code', 'value': 'bunny_010_0010'}
+    #
+    # For an item in the ShotgunHierarchyModel tree structure, this data is None.
+    #
+    field_data = shotgun_model.get_sanitized_data(item, shotgun_model.ShotgunModel.SG_ASSOCIATED_FIELD_ROLE)
+
+    if field_data is None and \
+       isinstance(sg_data, dict) and "has_children" in sg_data and "ref" in sg_data and \
+       isinstance(sg_data["ref"], dict) and "value" in sg_data["ref"]:
```

Comment:
To avoid hardcoding this knowledge of the dict structure in the conditional, could you check the model type via `isinstance(item.index().model(), ...)`?